web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```
@@ -0,0 +1,342 @@
+        TelemetryItem expectedTelemetry3 = sendGet(uri3, requestId3);
+
+        ArrayList<TelemetryItem> expectedTelemetries = new ArrayList<TelemetryItem>();
+        expectedTelemetries.add(expectedTelemetry1);
+        expectedTelemetries.add(expectedTelemetry2);
+        expectedTelemetries.add(expectedTelemetry3);
+
+        return expectedTelemetries;
+    }
+
+    /**
+     * Sends HTTP GET request and returns the expected telemetry from AppInsights
+     * @param uri The URI for the request
+     * @return The expected telemetry items from AppInsights
+     * @throws Exception
+     */
+    private TelemetryItem sendGet(URI uri, String requestId) throws Exception {
+        HttpURLConnection connection = (HttpURLConnection) uri.toURL().openConnection();
+
+        // optional default is GET
+        connection.setRequestMethod("GET");
+
+        //add request header
+        connection.setRequestProperty("User-Agent", config.getUserAgent());
+
+        System.out.println("Sending 'GET' request to URL: " + uri.toString());
+        Integer responseCode = connection.getResponseCode();
+        System.out.println("Response Code : " + responseCode);
+
+        // Create expected result
+        TelemetryItem expectedResult = new TelemetryItem(DocumentType.Requests, requestId);
+        expectedResult.setProperty("port", Integer.toString(uri.getPort()));
+        expectedResult.setProperty("responseCode", responseCode.toString());
+        expectedResult.setProperty("uri", uri.toString());
+        expectedResult.setProperty("userAgent", config.getUserAgent());
+
+        try {
+            String[] params = uri.getQuery().split("&");
+            for (String param : params) {
+                String name = param.split("=")[0];
+                String value = param.split("=")[1];
+                expectedResult.setProperty("queryParameter." + name, value);
+            }
+        }
+        catch (Exception e) {
```

In what scenarios are we expecting this to fail and should such failures be ignored?